tests/cupy_tests/linalg_tests/test_decomposition.py

```diff
@@ -15,14 +15,26 @@ class TestCholeskyDecomposition(unittest.TestCase):
 
     @testing.for_float_dtypes(no_float16=True)
```

Comment:
What happens when float16 array is given? How about casting an input array to float32 and casting a return value to float16? If you don't want to support float16, insert type check in the method.